In [1]:
import math
import numpy as np
import pandas as pd
from scipy import stats as st

Все методы так же можно посмотреть в этой [статьи](https://100task.ru/sample/70.aspx).

#### Метод центральной функции

Пусть $ G(X, \theta) $ - это функция, распределение которой известно, но распределение этой функции не зависит от $ \theta $.

Где:

$ G(X, \theta) $ - функция является сл. вел., т.к. зависит от конкретной реализайии выборки $X.

$X$ - выборка.

Например:

$ G(X, \theta) = \sqrt{n} \frac{\overline x - \theta}{\sigma} \sim N(0,1) $, видим, функция зависит от $ \theta $, а распределение этой функции от $ \theta $ не зависит!

Возьмем $ \alpha_1, \alpha_2 \in (0, 1)  $, т.ч. $ \alpha_2 - \alpha_1 = \alpha

и $g_i$ - это $\alpha_i$ квантиль распределения функции $ G(X, \theta) $.

Тогда $S(x) = (\theta \in \Theta | q_1 \leq G(X, \theta) \leq q_2)$  - это доверительная область для $\theta$, уровень доверия $\alpha$

Действительно

$ P_\theta(\theta \in S(x)) = P_\theta(q_1 \leq G(X, \theta) \leq q_2) = \alpha_2 - \alpha_1 = \alpha  $

#### Интервал для $a$, если $\sigma$ известно

Пусть есть выборка $X = (x_1, ..., x_n) \sim N(\theta, \sigma^2)$, где $\sigma$ известно. Построить дов. интервал для $\theta$

Заметим, что $ x_i - \theta \sim N(0, \sigma^2)$. Воспользуемся всей выборкой $ \overline x - \theta \sim N(0, \frac{\sigma^2}{n}) $ и усредним дисперсию. Вынесем из нормального распределения дисперсию 

$ \sqrt{n} \frac{\overline x - \theta}{\sigma} \sim N(0, 1) $

Получили центральную функцию

$ G(X, \theta) = \sqrt{n} \frac{\overline x - \theta}{\sigma} $

Обозначение: $ z_p - p$- квантиль $N(0,1)$

Ограничим центральную функцию квантилями

$ P_\theta( - z_\frac{1+\alpha}{2} \leq \sqrt{n} \frac{\overline x - \theta}{\sigma} \leq z_\frac{1+\alpha}{2} ) = \alpha $

либо

$ P_\theta( z_\frac{\alpha}{2} \leq \sqrt{n} \frac{\overline x - \theta}{\sigma} \leq z_{1 - \frac{\alpha}{2}} ) = 1 - \alpha $

Выразим отсюда $\theta$

$ P_\theta( \overline x - z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x + z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}}) = \alpha $

либо

$ P_\theta( \overline x - z_{1 - \frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x - z_\frac{\alpha}{2} \frac{\sigma}{\sqrt{n}}) = 1 - \alpha $

**Ответ:** 

Интервал

$(\overline x \pm z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}})$

либо

$ (\overline x - z_{1 - \frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}}, \overline x - z_\frac{\alpha}{2} \frac{\sigma}{\sqrt{n}}) $

#### Интервал через функцию Лапласа (стандартное норм. распределение)

Метод из этой [статьи](https://infopedia.su/5x215b.html).

Рассмотрим $\overline X$ как сл. вел. у которой $$E(\overline X) = E(\frac{1}{n} \sum\limits_{i=1}^n x_i) = \frac{1}{n} E \sum\limits_{i=1}^n x_i = a$$ т.к. $x_i$ имеют одинаковый закон распр. как исходная генеральная совокупность, следовательно имеют одинаковое мат. ож., такое же как мат. ож. ген. совокупности. 

$$D(\overline X) = D(\frac{1}{n} \sum\limits_{i=1}^n x_i ) = \frac{1}{n^2} D (\sum\limits_{i=1}^n x_i) = \frac{1}{n^2} \sum\limits_{i=1}^n Dx_i = \frac{1}{n^2} \cdot n \cdot Dx_1 = \frac{Dx_1}{n}$$

[Свойства дисперсии](http://ecocyb.narod.ru/513/matmod/mms2_6.htm)

$$ \sigma (\overline X) = \sqrt{\frac{Dx_1}{n}} = \frac{\sigma}{\sqrt{n}}$$

По следствию [интегральной теоремы Лапласа](http://matica.org.ua/metodichki-i-knigi-po-matematike/kurs-vysshei-matematiki-4/23-funktciia-laplasa)

$$ P(|\hat{\theta} - \theta| \leq \Delta) = \overline Ф \bigg (\frac{\theta + \Delta - \theta}{\sigma(\hat{\theta})} \bigg ) - \overline Ф \bigg (\frac{\theta - \Delta - \theta}{\sigma(\hat{\theta})} \bigg ) = \overline Ф \bigg (\frac{\Delta}{\sigma(\hat{\theta})} \bigg ) - \overline Ф \bigg (-\frac{\Delta}{\sigma(\hat{\theta})} \bigg ) = 2 \overline Ф \bigg (\frac{\Delta}{\sigma(\hat{\theta)}} \bigg ) = 2 \overline Ф(t)$$

где $\Delta$ - точность оценки, $\sigma(\hat{\theta}) = \sqrt{D(\hat{\theta})}$ - отклонение оценки, $t$ - квантиль, $\overline Ф$ - нормированная функция Лапласа, чтобы её вычислить, можно от функции Лапаласа вычесть 0.5, т.е. $ \overline Ф(x) = Ф(x) - 0.5$ подробнее в самом низу [здесь](https://ru.stackoverflow.com/questions/542603/Как-получить-значение-функции-лапласа-python).

В нашем случае имеем

$$ P(|\overline X - a| \leq \Delta) = 2Ф(\frac{\Delta}{\frac{\sigma}{\sqrt{n}}}) = 2Ф(t)$$

где $t = \frac{\Delta \sqrt{n}}{\sigma}$, откуда $ \Delta = \frac{\sigma t}{\sqrt{n}}$ - точность оценки. Получаем

$$\overline X - \delta \leq a \leq \overline X + \delta$$

$$\overline X - \frac{\sigma t}{\sqrt{n}} \leq a \leq \overline X + \frac{\sigma t}{\sqrt{n}}$$

где $t$ - это квантиль стандартного нормального распределения!

##### Задача №1. $\sigma$ известно, $a$ неизвестно

Пусть сл. вел. рост мужчин. Построить 95% дов. интервал для среднего роста мужчин. Будем предполагать, что сл. вел. $\xi \sim N(a, \sigma^2)$, $\sigma$ = 7. $a$ не известно.

**Решение:**

В качестве точечной оценки мат. ож. возьмем выборочное среднее $\overline x_{выб}$, т.к. генеральная совокупность имеет норм. закон распределения $N(a, \sigma)$, то выборочное среднее будет так же иметь нормальный закон распределения $ \overline x_{выб} \sim N( a, \frac{\sigma}{\sqrt{n}} ) $

Построим статистику $Y$, центрируем и нормируем её.

$ Y = \frac{\overline x_{выб} - a}{\frac{\sigma}{\sqrt{n}}} $, тогда $Y \sim N(0,1) $, при этом $Y(\overline x_{выб}, a)$ как функция от статистики $\overline x_{выб}$ и $a$ будет непрерывна и будет строго убывать по $a$.

Зададим $\alpha = 0.95$, вероятность с которой мы хотим быть уверены, что значение будет лежать в найденном интервале и найдём квантили стандартного норм. распр. $N(0,1)$ порядка $z_{\frac{1+\alpha}{2}}$.

$ P(- z_{\frac{1 + \alpha}{2}} < \frac{\overline x_{выб} - a}{\frac{\sigma}{\sqrt{n}}}< z_{\frac{1 + \alpha}{2}} ) = P(\overline x_{выб} - z_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}}  <a< \overline x_{выб} + z_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}} ) $

либо 

Зададим $\alpha = 0.05$ и найдём квантили порядка $z_\frac{\alpha}{2}$ и $z_{1 - \frac{\alpha}{2}}$

$ P_\theta( z_\frac{\alpha}{2} \leq \sqrt{n} \frac{\overline x - \theta}{\sigma} \leq z_{1 - \frac{\alpha}{2}} ) = P_\theta( \overline x - z_{1 - \frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x - z_\frac{\alpha}{2} \frac{\sigma}{\sqrt{n}}) $

Построем оценку по выборке

In [ ]:
s = np.array([174.83, 172.75, 158.51, 182.70, 175.53, 161.66, 182.38, 185.49, 174.96, 172.73, 180.90, 169.08, 179.66, 174.01, 166.57, 168.18, 163.92, 174.85, 180.21, 176.37])
xs = s.mean()

sigma, n = 7, 20

Найдём квантиль

In [ ]:
# уровень доверия интервала
alpha = 0.95
# порядок квантиля z, крайняя правая точка доверительного интервала
p = (1 + alpha) / 2
print(p)
# z - квантиль порядка p, такое число, что данная случайная величина попадает левее его с вероятностью, не превосходящей p.
z = st.norm.ppf(p)
print(z)

0.975
1.959963984540054


Т.к. распределение симметричное, то 

$ z_{\frac{1 - \alpha}{2}} = - z_{\frac{1 + \alpha}{2}}$ 

$ z_{\frac{1 - \alpha}{2}} = - 1.96 $

либо

$ u_{1 -\frac{\alpha}{2}} = u_{-\frac{\alpha}{2}} \approx 1.96$

$ u_{\frac{\alpha}{2}}  \approx -1.96 $

Решим уравнение

$ P_\theta( \overline x - z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x + z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}}) = \alpha $

либо

$ P(\overline x_{выб} - u_{\frac{1-\alpha}{2}} \frac{\sigma}{\sqrt{n}}  <a< \overline x_{выб} - u_{\frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}} ) = 1- \alpha$

In [ ]:
sd = xs - (z * sigma/math.sqrt(n))
su =  xs + (z * sigma/math.sqrt(n))

print(f'{sd:.2f}, {su:.2f}')

170.70, 176.83


либо

In [ ]:
sd = xs - (z * sigma/math.sqrt(n))
su =  xs - (-z * sigma/math.sqrt(n))

print(f'{sd:.2f}, {su:.2f}')

170.70, 176.83


**Ответ:**

$ P( 170.70  <a< 176.83 ) = 0.95 $

##### Задача №2. $\sigma$ известно, $a$ неизвестно

Найти доверительный интервал для оценки математического ожидания a
**нормального распределения** с надежностью 0,95, зная выборочную среднюю x , объем
выборки n и среднее квадратическое отклонение $ \sigma $.

$ \overline x = 75.12 $, $ n = 121 $, $ \sigma = 11 $

**Решение:**

В качестве точечной оценки мат. ож. возьмем выборочное среднее $\overline x_{выб}$ Построим центральную функцию $G$

$ G(X, a) = \frac{\overline x - a}{\frac{\sigma}{\sqrt{n}}} \sim N(0,1) $

Положим $\alpha = 0.95$

Найдём квантиль $z_{\frac{1+\alpha}{2}}$



 

In [ ]:
alpha = 0.95
p = (1 + alpha) / 2
z = st.norm.ppf(p)
round(z, 2)

1.96

Решим уравнение

$ P_\theta( \overline x - z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x + z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}}) = \alpha $

In [ ]:
xs = 75.12
n = 121
sigma = 11
sd = xs - (z * sigma / math.sqrt(n))
su = xs + (z * sigma / math.sqrt(n))

print(f'{sd:.2f}, {su:.2f}')

73.16, 77.08


**Ответ:**

$ P( 73.16  < a < 77.03 ) = 0.95 $

##### Задача №3. $\sigma$ известно, $a$ неизвестно

На основе продолжительных наблюдений за весом Х г. пакетов с орешками, заполняемых автоматически, установлено, что среднее квадратическое отклонение веса пакетов равно 10г. Средний вес 100 наудачу выбранных пакетов оказался равным 249 г. Укажите доверительный интервал для среднего веса пакетов, построенный с доверительной вероятностью 0.9. Предполагается, что распределение веса пакетиков подчиняется нормальному закону распределения.

In [ ]:
sigma, xs, n = 10, 249, 100

**Решение:**

В качестве точечной оценки мат. ож. возьмем выборочное среднее $\overline x_{выб}$ 

$ G(X, a) = \frac{\overline x - a}{\frac{\sigma}{\sqrt{n}}} \sim N(0,1) $

Положим $\alpha = 0.9$

Найдём квантиль $z_{\frac{1+\alpha}{2}}$

In [ ]:
alpha = 0.9
p = (1 + alpha) / 2
z = st.norm.ppf(p)
round(z, 2)

1.64

Решим уравнение

$ P_\theta( \overline x - z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}} \leq \theta \leq \overline x + z_\frac{1+\alpha}{2} \frac{\sigma}{\sqrt{n}}) = \alpha $

In [ ]:
sd = xs - (z * sigma / math.sqrt(n))
su = xs + (z * sigma / math.sqrt(n))

print(f'{sd:.3f}, {su:.3f}')

247.355, 250.645


##### Задача №4 $\sigma$ известно, $a$ неизвестно

По данным 7 измерений некоторой величины найдены средняя результатов
измерений, равная 30 и выборочная дисперсия, равная 36. Найдите границы, в которых с надежностью 0,99 заключено истинное значение измеряемой величины. 

**Решение:**

Истинное значение измеряемой величины равно ее математическому ожиданию. Поэтому задача сводится к оценке математического ожидания (при неизвестном а) при помощи доверительного интервала. Распределение нам не известно. Предположим, что результаты измерений подчиняются нормальному закону распределения. Тогда интервал будет

$ P_\theta ( \overline X - z_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}} \leq a \leq \overline X + z_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}}) = \alpha$

In [ ]:
xs = 30
sigma = math.sqrt(36)
n = 7
alpha = 0.99
p = (1 + alpha) / 2
z = st.norm.ppf(p)

sd = xs - (z * (sigma/math.sqrt(n)))
su = xs + (z * (sigma/math.sqrt(n)))

print(f'{sd:.4f}, {su:.4f}')

24.1586, 35.8414


#### Интервал для $\sigma$, если $a$ известно

Вспомним, если $ \xi \sim N(0, 1) $, то $ \xi^2_1 + ... + \xi^2_k \sim  \chi^2_k$. Тогда

$ \frac{x_i - a}{\sigma} \sim N(0,1) $, то

$ \sum\limits_{i=1}^k \bigg ( \frac{ x_i - a}{\sigma} \bigg )^2 \sim \chi^2_k $ 

Определим центральную функцию

$ G(X, \theta) = \sum\limits_{i=1}^k \bigg ( \frac{ x_i - a}{\sigma} \bigg )^2 $ 

$ \chi^2_{k, p} - p $ - квантиль распределения $ \chi^2_k $

Возьмем квантили для ограничение дов. интервала 

$ \chi^2_{k, \frac{1-\alpha}{2}} $ и $ \chi^2_{k, \frac{1+\alpha}{2}} $

Получим

$ P_\theta \bigg ( \chi^2_{k, \frac{1-\alpha}{2}} \leq \frac{1}{\sigma^2} \sum\limits_{i=1}^k(x_i - a)^2 \leq \chi^2_{k, \frac{1+\alpha}{2}} \bigg ) = \alpha $

Выразим $ \sigma $

$ P_\theta \bigg ( \sqrt{ \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1+\alpha}{2}}} } \leq \sigma \leq \sqrt{ \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1-\alpha}{2}}}} \bigg ) = \alpha $

Выразим $\sigma$ получаем интервал:

$ \bigg ( \sqrt{ \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1+\alpha}{2}}} }, \sqrt{ \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1-\alpha}{2}}}} \bigg ) $


##### Задача №1. $a$ известно, $\sigma$ неизвестно

Постройте 95%-доверительный интервал для дисперсии $\sigma^2$ ошибок измерений, если  предполагается, что ошибки измерений подчиняются нормальному закону с мат.ожиданием $a=0$ и неизвестной дисперсией $\sigma^2$.

In [ ]:
a, alpha = 0, 0.95
s = np.array([-0.07, -0.74, 2.82, 0.19, 0.67, 0.79, 1.83, -0.14, -0.74, 1.79, 0.48, 0.68, -0.69, 0.58, 1.22, -0.68, 0.73, 1.4, 0.41, 0.66])
df = len(s)
xs = s.mean()

Найдём квантили $ \chi^2_{k, \frac{1+\alpha}{2}} $ и $ \chi^2_{k, \frac{1-\alpha}{2}} $

In [ ]:
p1 = (1 + alpha) / 2
p2 = (1 - alpha) / 2
xi_1 = round(st.chi2.ppf(round(p1, 3), df=df), 4)
xi_2 = round(st.chi2.ppf(round(p2, 3), df=df), 4)

In [ ]:
# можно так
xi_2, xi_1 = st.chi2.interval(alpha, df)

Найдем интервал $ \bigg ( \ \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1+\alpha}{2}}} \leq \sigma^2 \leq \frac{\sum\limits_{i=1}^k(x_i - a)^2}{\chi^2_{k, \frac{1-\alpha}{2}}}  \bigg ) $

In [ ]:
sd = math.sqrt(sum((s - a)**2) / xi_1)
su = math.sqrt(sum((s - a)**2) /xi_2)

print(f'{sd:.3f}, {su:.3f}')

0.826, 1.558


In [ ]:
sd = sum((s - a)**2) / xi_1
su = sum((s - a)**2) /xi_2

print(f'{sd:.3f}, {su:.3f}')

0.826, 1.558


**Ответ:**

$ P_\theta \bigg ( 0.826 \leq \sigma^2 \leq 1.558  \bigg ) = \alpha$

#### Интервал для $a$, если $a$ и $\sigma$ неизвестны

[Статья](https://habr.com/ru/company/stepic/blog/311354/) про кол-во степеней свободы $k = n-1$

Пусть данна выборка $X = (x_1, ..., x_n)$ из $ X \sim N(a, \sigma)$, $a$ и $\sigma^2$ неизвестны.

В качестве точечной оценки дисперсии:

$ s^2 = \frac{1}{n-1} \sum\limits_{i=1}^n (x_i - \overline X)^2  $

Для построения дов. интервал **для мат. ож.** построим сл. вел.

$\xi = \sqrt{n} \frac{\overline X - a}{\sigma} \sim N(0,1)$

$ \eta = \frac{n s^2}{\sigma^2} \sim \chi^2_{n-1}$



Воспользуемся свойством распределения Стьюдента. Если $\xi \sim N(0,1)$ и $\eta \sim \chi^2_k$ независимы, то $ \zeta = \frac{\xi}{\sqrt{\frac{\eta}{k}}} \sim T_k $.

Количество степеней свободы $k$ в нашем случае равно $n-1$. Построим центральную функцию $G(X, a)$

$ G(X, a) = \frac{ \sqrt{n} \frac{\overline X - a}{\sigma} }{ \sqrt{ \frac{ns^2}{\sigma^2}  \frac{1}{n-1} } } = \sqrt{n-1} \frac{\overline X - a}{s} \sim T_{n-1}$ 

Построим интервал

$ P_\theta ( -t_{n-1, \frac{1+\alpha}{2}} \leq \sqrt{n-1} \frac{\overline X - a}{s} \leq  t_{n-1, \frac{1+\alpha}{2}} ) = \alpha $

$ P_\theta ( \overline X - t_{n-1, \frac{1+\alpha}{2}} \frac{s}{\sqrt{n-1}} \leq a \leq   \overline X + t_{n-1, \frac{1+\alpha}{2}} \frac{s}{\sqrt{n-1}} ) = \alpha $

Получили интервал

$ \bigg ( \overline X \pm t_{n-1, \frac{1+\alpha}{2}} \frac{s}{\sqrt{n-1}}  \bigg )$

##### Задача №1. интервал для $a$, где $a$ и $\sigma^2$ неизвестно 

По данным 7 измерений некоторой величины найдены средняя результатов
измерений, равная 30 и выборочная дисперсия, равная 36. Найдите границы, в которых с надежностью 0,99 заключено истинное значение измеряемой величины. 

In [ ]:
xs = 30
sigma = math.sqrt(36)
n = 7
alpha = 0.99
p = (1 + alpha) / 2
t = st.t.ppf(p, df=n-1)

sd = xs - (t * (sigma/math.sqrt(n-1)))
su = xs + (t * (sigma/math.sqrt(n-1)))

print(f'{sd:.2f}, {su:.2f}')

20.92, 39.08


**Ответ:**

$(20.92 \leq a \leq 39.08) $



##### Задача №2. Интервал для $a$ и $\sigma$, $a$ и $\sigma$ неизвестны

По группе семей с доходом 154 руб./чел. зафиксированы следующие цифры
потребления молока за месяц (на одного человека) см. T1. Найти доверительный интервал для математического ожидания и дисперсии с надежностью γ = 0,95 , дать точность оценки. Выборка произведена из нормальной совокупности. 


In [ ]:
T1 = pd.DataFrame({'x_i': [8.3, 8.6, 8.7, 8.8, 9.1, 9.3, 9.4, 13.4, 13.5, 13.8, 13.9, 14.1, 14.3]}).T
T1

0    1    2    3    4    5    6     7     8     9    10    11    12
x_i  8.3  8.6  8.7  8.8  9.1  9.3  9.4  13.4  13.5  13.8  13.9  14.1  14.3

Для построения дов. инт. для мат. ож. воспользуемся формулой:

$\overline X - t_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}} \leq a \leq \overline X + t_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}}$

Вычислим выборочные характеристики:

$\overline X = \frac{1}{n} \sum\limits_{i=1}^n x_i$

$D_x = \frac{1}{n} \sum\limits_{i=1}^n (x_i - \overline X)^2 $ 

$se = \sqrt{D_x}$

In [ ]:
xs = T1.values.mean()
D = T1.values.var()
se = math.sqrt(D)

print(f'mean: {xs:.3f}, var: {D:.3f}, se: {se:.3f}')

mean: 11.169, var: 6.202, se: 2.490


Найдём квантиль Стьюдента $ t_{\frac{1 + \alpha}{2}} $

In [ ]:
n = T1.shape[1]
alpha = 0.95
p = (1 + alpha) / 2
t = st.t.ppf(p, df=n-1)
print(f'quantile: {t:.3f}')

quantile: 2.179


Подставим в формулу $\overline X - t_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}} \leq a \leq \overline X + t_{\frac{1 + \alpha}{2}} \frac{\sigma}{\sqrt{n}}$

In [ ]:
sd = xs - t * se / math.sqrt(n)
su = xs + t * se / math.sqrt(n)

print(f'{sd:.2f}, {su:.2f}')

9.66, 12.67


Далее необходимо вычислить интервал для дисперсии. Вначале найдём интервал для стандартного отклонения, а потом возведем интервал в квадрат и получим интервал для дисперсии. Мы знаем формулу интервала для стандартного отклонения:

$ \sqrt{ \frac{n s^2}{\chi^2_{n-1, \frac{1+\alpha}{2}}} } \leq \sigma \leq \sqrt{ \frac{n s^2}{\chi^2_{n-1, \frac{1-\alpha}{2}}} } $



In [ ]:
chi_left = math.sqrt( n * D / st.chi2.ppf(p, df=n-1) )
chi_right = math.sqrt( n * D / st.chi2.ppf((1 - alpha)/2, df=n-1) )

print(f'left spacing: {chi_left:.3f}, right spacing: {chi_right:.3f}')

left spacing: 1.859, right spacing: 4.279


А интервал для дисперсии будет 

In [ ]:
print(f'left spacing: {chi_left**2:.3f}, right spacing: {chi_right**2:.3f}')

left spacing: 3.455, right spacing: 18.309


**Ответ:**

$ 9.66 \leq a \leq 12.67 $

$ 1.86 \leq \sigma \leq 4.28 $

$ 3.45 \leq \sigma^2 \leq 18.3 $

#### Ассимптотический доверительный интервал

Пусть $ X = (x_1, ..., x_n) $ - выборка неограниченного размера из неизвестного распределения $ P \in (P_\theta | \theta \in \Theta) $.

Посл-ть пар статистик $\bigg (T^{(n)}_1 (x_1, ..., x_n) \leq \theta \leq T^{(n)}_2 (x_1, ..., x_n) \bigg )$ называется ассимптотическим дов. интервалом уровня доверия $\alpha$ для параметра $\theta$, если $ \forall\theta \in \Theta : \lim\limits_{n \to \infty} inf P_\theta \bigg ( T^{(n)}_1 (x_1, ..., x_n) \leq \theta \leq T^{(n)}_2 (x_1, ..., x_n) \bigg ) \geq \alpha $.

#### Метод построения ассимптотического дов. интервала

1. Пусть $\hat{\theta}$ - а.н.о $\theta$ с ас. дисп. $\sigma^2(\theta)$ запишем это условие в виде формулы

$\sqrt{n} (\hat{\theta} - \theta) \xrightarrow{d_\theta} N(0, \sigma^2(\theta)) $

2. Поделим всё на $\sigma(\theta)$:

$\sqrt{n} \frac{(\hat{\theta} - \theta)}{\sigma(\theta)}  \xrightarrow{d_\theta} N(0, 1)$

$ P_\theta ( \sqrt{n} \frac{|\hat{\theta} - \theta|}{\sigma(\theta)} ) \leq z_{\frac{1+\alpha}{2}}) \rightarrow \alpha $

Проблема в том, что мы не знаем $\theta $, а значит не знаем $\sigma(\theta)$. Поэтому

3. Пусть $ \hat{\sigma} $ - состоятельная оценка $ \sigma(\theta) $. Заменим

$ \sqrt{n} \frac{\hat{\theta} - \theta}{\hat{\sigma}} = \sqrt{n} \frac{(\hat{\theta} - \theta)}{\sigma(\theta)} \cdot \frac{\sigma(\theta)}{\hat{\sigma}} \xrightarrow{d_\theta} N(0, 1) $

4. $ P_\theta ( -z_{\frac{1+\alpha}{2}} \leq \sqrt{n} \frac{(\hat{\theta} - \theta)}{\hat{\sigma}} ) \leq z_{\frac{1+\alpha}{2}}) \rightarrow \alpha $

$ P_\theta ( -z_{\frac{1+\alpha}{2}} \hat{\sigma} \leq \sqrt{n} (\hat{\theta} - \theta) \leq z_{\frac{1+\alpha}{2}} \hat{\sigma} ) \rightarrow \alpha $

$ P_\theta ( -z_{\frac{1+\alpha}{2}} \frac{\hat{\sigma}}{\sqrt{n}} \leq \hat{\theta} - \theta \leq z_{\frac{1+\alpha}{2}} \hat{\sigma} ) \rightarrow \alpha $

$ P_\theta ( \hat{\theta} - z_{\frac{1+\alpha}{2}} \frac{\hat{\sigma}}{\sqrt{n}} \leq \theta \leq \hat{\theta} + z_{\frac{1+\alpha}{2}} \hat{\sigma} ) \rightarrow \alpha $

Получаем интервал

$ (\hat{\theta} \pm z_{\frac{1+\alpha}{2}} \frac{\hat{\sigma}}{\sqrt{n}})$

5. Откуда взять $\hat{\sigma}$ ?

$\hat{\theta} $ - а.н.о. $\theta \Rightarrow \hat{\theta} $ - состоятельная оценка $\theta$

Тогда если $\sigma(\theta)$ непрерывна, то по теоре. о наследовании сходимости $ \hat{\sigma} = \sigma(\hat{\theta}) $ - состоятельная оценка $\sigma(\theta)$.

##### Задача №1

Большая партия изделий может содержать некоторую долю дефектных. Поставщик утверждает, что эта доля составляет 5%, покупатель  предполагает, что доля дефектных изделий равна 10%.

По предложенной выборке постройте 95%-й приближенный доверительный интервал  для доли брака в партии  и ответьте на вопрос: с большей степенью уверенности можно утверждать, что доля брака  в партии больше 5%? (G - доля брака больше 5%, L - в противном случае).

**Решение:**

Рассмотрим схему Бернулли, в которой n испытаний.

Выборка $X = x_1, ..., x_n$ состоит из последоватеьлности нулей и единиц.

Пусть $m$ - число успехов.

Точечной оценкой параметра $p$ будет $\hat{p} = \frac{m}{n}$

$\hat{p}$ ассимптотически нормальна:

по теории

$\sqrt{n} (\hat{\theta} - \theta) \xrightarrow{d_\theta} N(0, \sigma^2(\theta)) $

в распределении Бернулли дисперсия равна 

$\sigma^2(p) = p(1-p)$

тогда оценка равна

$\sigma^2(\hat{p}) = \frac{m}{n}(1 - \frac{m}{n}) $

$ \sqrt{n} (\frac{m}{n} - p) = \frac{m-np}{\sqrt{n}} = \xrightarrow[n \rightarrow \infty ]{d_\theta} \zeta \sim N(0, \sqrt{p(1-p)})$

Тогда имеет место сходимость

$ \frac{ \sqrt{n} (\frac{m}{n} - p) }{ \sqrt{ \frac{m}{n}(1 - \frac{m}{m}) } } \xrightarrow[n \rightarrow \infty ]{d_\theta} \eta \sim N(0,1) $ 

Получаем

$ P_\theta ( - z_{\frac{1+\alpha}{2}} \leq \frac{ \sqrt{n} (\frac{m}{n} - p) }{ \sqrt{ \frac{m}{n}(1 - \frac{m}{m}) } } \leq z_{\frac{1+\alpha}{2}} ) = \alpha$

Искомый интервал будет

$ \bigg ( \frac{m}{n} \pm z_{\frac{1+\alpha}{2}} \frac{\sqrt{ \frac{m}{n}(1 - \frac{m}{m}) }}{\sqrt{n}} \bigg ) $


In [ ]:
s = np.array([0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0])

In [ ]:
m = sum(s)
n = len(s)
p_hat = m/n
sigma = math.sqrt(p_hat * (1 - p_hat))

In [ ]:
alpha = 0.95
p = (1 + alpha) / 2
z = st.norm.ppf(p)

In [ ]:
sd = p_hat - z * (sigma/ math.sqrt(n))
su = p_hat + z * (sigma/ math.sqrt(n))

print(f'{sd:.4f}, {su:.4f}', 'L' if su - sd / 2  < 0.05 else 'G')

0.1224, 0.1856 G


**Ответ:**

$ P_\theta \bigg ( 0.1224 \leq \sigma \leq 0.1856  \bigg ) = \alpha$

##### Задача №2

В 10000 сеансах игры с автоматом выигрыш появился 4000 раз. Найдите 95% доверительный интервал для вероятности выигрыша.

**Решение:**

Мы имеем распределение Бернулли. Используя знания из предыдущей задачи, искомый интервал будет

$ \bigg ( \frac{m}{n} \pm z_{\frac{1+\alpha}{2}} \frac{\sqrt{ \frac{m}{n}(1 - \frac{m}{m}) }}{\sqrt{n}} \bigg ) $, где

$\hat{p} = \frac{m}{n}$ - оценка вероятности.

$m$ - число успехов, $n$ - число испытаний

$n$ - 10000, $m$ - 4000, $z_{\frac{1+\alpha}{2}}$ - 1.96



In [ ]:
n, m, z = 10000, 4000, 1.96
p_hat = m / n 

In [ ]:
sd = p_hat - (z * (math.sqrt(p_hat * (1 - p_hat)) / math.sqrt(n)))
su = p_hat + (z * (math.sqrt(p_hat * (1 - p_hat)) / math.sqrt(n)))


print(f'{sd:.3f}, {su:.3f}')

0.390, 0.410


**Ответ:**

$ P_\theta \bigg ( 0.39 \leq \sigma \leq 0.41  \bigg ) = \alpha$

#### Оценка доли по повторной и бесповторной выборке. Оценка вероятности!

Материал из [статьи](http://www.mathprofi.ru/ocenki_po_povtornoy_i_bespovtornoy_vyborke.html).

Пусть из нормально распределенной (или около того) генеральной совокупности объёма $N$ проведена выборка объёма $n$ и по её результатам найдена выборочная средняя $\overline X_в$ и исправленная выборочная дисперсия $s^2$.

Тогда доверительный интервал для оценки генеральной средней $\overline X_г = a$ имеет вид:

$ \overline X_в - \Delta < \overline X_г < \overline X_в + \Delta $, где $\Delta$ - точность оценки, которую также называют предельной ошибкой репрезентативности выборки.

Точность оценки рассчитывается как произведение $\Delta = t \cdot \mu$  – коэффициента доверия $t$ на среднюю ошибку выборки $\mu$.

Для бесповторной выборки она составляет

$$\mu = \sqrt{ \frac{s^2}{n} ( 1 - \frac{n}{N} ) }$$

для повторной

$$\mu = \sqrt{ \frac{s^2}{n} }$$

В том случае, если изначально известна генеральная дисперсия $\sigma^2$, то используют, конечно, её.

Если объём выборки $n<30$, то коэффициент доверия определяется с помощью распределения Стьюдента с $k$ степенями свободы, где $k = n- 1$.

Если $n>30$, то чаще пользуются соотношением $2Ф(t) = \alpha$, где $Ф(t)$ – функция Лапласа, а $\alpha$ – доверительная вероятность.

Значение «альфа» показывает, с какой вероятностью построенный интервал $ \overline X_в \pm \Delta $  накрывает истинное значение $\overline X_г$.

##### Задача №1

Из партии объемом 500 однородных товаров для проверки по схеме случайной
бесповторной выборки отобрано 70 товаров, среди которых оказалось 56 небракованных.
Найдите вероятность того, что доля бракованных товаров во всей партии отличается от полученной доли в выборке не более чем на 0,02 (по абсолютной величине), а также границы, в которых с надежностью 0,96 заключена доля бракованных товаров во всей партии. 

Чтобы найти вероятность того, что доля бракованных товаров во всей выборке отличается от полученной доли, воспользуемся следствием интегральной теоремы [Лапласа](http://matica.org.ua/metodichki-i-knigi-po-matematike/kurs-vysshei-matematiki-4/23-funktciia-laplasa)

$$ P(|\hat{\theta} - \theta| \leq \Delta) = \overline Ф \bigg (\frac{\theta + \Delta - \theta}{\sigma(\hat{\theta})} \bigg ) - \overline Ф \bigg (\frac{\theta - \Delta - \theta}{\sigma(\hat{\theta})} \bigg ) = \overline Ф \bigg (\frac{\Delta}{\sigma(\hat{\theta})} \bigg ) - \overline Ф \bigg (-\frac{\Delta}{\sigma(\hat{\theta})} \bigg ) = 2 \overline Ф \bigg (\frac{\Delta}{\sigma(\hat{\theta)}} \bigg ) = 2 \overline Ф(t)$$

где $\Delta$ - точность оценки, $\sigma(\hat{\theta}) = \sqrt{D(\hat{\theta})}$ - отклонение оценки, $t$ - квантиль, $\overline Ф$ - нормированная функция Лапласа, чтобы её вычислить, можно от функции Лапаласа вычесть 0.5, т.е. $ \overline Ф(x) = Ф(x) - 0.5$ подробнее в самом низу [здесь](https://ru.stackoverflow.com/questions/542603/Как-получить-значение-функции-лапласа-python).

$t$ - квантиль, можно вывести из формулы точности $\Delta = t \cdot \mu$, где $\mu$ - средняя ошибка выборки, для бесповторной выборки равна $$\mu = \sqrt{ \frac{s^2}{n} ( 1 - \frac{n}{N} ) }$$

где $s^2$ - исправленная выборочная дисперсия, если есть, то нужно использовать $\sigma^2$ стандартное отклонение генеральной совокупности, $N$ - кол-во товаров генеральной совокупности, $n$ - кол-во выбранных товаров.

$\sigma^2 = D$, дисперсия распределения Бернули равна $D = pq = p(1-p)$, где $p$ - доля бракованных товаров в выборке.

$\hat{p} = \frac{70 - 56}{70} = \frac{14}{70} = 0,2$

По условию предельная ошибка, не превосходит $\Delta \leq 0.02$, составим уравнение, откуда выразим искомый параметр $t$.

$\Delta = t \cdot \mu = t \cdot \sqrt{ \frac{\hat{p}(1-\hat{p})}{n} (1 - \frac{n}{N}) } = t \cdot \sqrt{ \frac{0,2(1-0,2)}{70} (1 - \frac{70}{500}) } \leq 0.02$

$t \cdot 0.044 \leq 0.02$

$t \leq 0,45$

Вычислим искомую вероятность, что доля бракованных товаров во всей партии отличается от полученной доли в выборке не более чем на 0,02.

$ P(|\hat{\theta} - \theta| \leq \Delta)  = P(|\hat{p} - p| \leq 0.02) = 2 \overline Ф(t) = 2 \overline Ф(0.45) = 0,35$

Тут нужно вычислить нормированную функцию Лапласа, [табличные](https://100task.ru/sample/119.aspx) данные как раз берутся из неё. Нормированная функция Лапласа связана с функцией Лапласа соотношением

$ \overline Ф(x) = Ф(x) - 0.5 $

In [ ]:
# cdf - плотность вероятности, интеграл ф-и Лапласа
F_normalized = st.norm.cdf(0.45, loc=0, scale=1) - 0.5
p =  2 * F_normalized
p

0.34728955942416007

Мы получили, что вероятность того, что доля бракованных товаров во всей партии отличается от полученной доли в выборке не более чем на 0,02 (по абсолютной величине) равна 0.35. 

Теперь необходимо найти границы, в которых с надежностью 0,96 заключена доля бракованных товаров во всей партии. Для этого вначале найдём абсолютное значение квантиля распределения Стьюдента.

In [ ]:
alpha = 0.96
t = (1 - alpha) / 2
t_quantile = abs(st.t.ppf(t, df=500-1))
t_quantile

2.059131485826451

Тогда предельная ошибка

$\Delta = t \cdot \mu = t \cdot \sqrt{ \frac{p(1-p)}{n} (1 - \frac{n}{N}) } = 2.05 \cdot \sqrt{ \frac{0,2(1-0,2)}{70} (1 - \frac{70}{500}) } = 0.091$

Границы доли бракованных товаров: 

$(\hat{p} - \Delta, \hat{p} + \Delta) = (0.2 - 0.091, 0.2 + 0.091) = (0.109, 0.291)$

**Ответ:**

Вероятность, что доля бракованных товаров во всей партии отличается от полученной доли в выборке не более чем на 0,02.

$P(|\hat{p} - p| \leq 0.02) = 0.35$

Границы, в которых с надежностью 0,96 заключена доля бракованных товаров во всей партии. 

$(\hat{p} - \Delta, \hat{p} + \Delta) = (0.109, 0.291)$

##### Задача №2

С целью размещения рекламы опрошено 420 телезрителей, из которых данную
передачу смотрят 170 человек. С доверительной вероятностью γ=0,91 найти долю
телезрителей, охваченных рекламой в лучшем случае. 

**Решение:**

Мы имеем распределение Бернулли, человек смотрел или не смотрел передачу. Так как человек может посмотреть передачу несколько раз, будем использовать среднюю ошибку повторной выборки

$$\mu = \sqrt{ \frac{s^2}{n} }$$

Интервал для вероятности будет

$ \hat{p} - t_{n-1, \frac{1 + \alpha}{2}} \sqrt{ \frac{\hat{p} (1-\hat{p})}{n} } \leq p \leq \hat{p} + t_{n-1, \frac{1 + \alpha}{2}} \sqrt{ \frac{\hat{p} (1-\hat{p})}{n} } $, где

$n$ - объем выборки

$ \hat{p} = \frac{170}{400} = 0.4048 $ - относительная частота.

$t_{n-1, \frac{1 + \alpha}{2}}$ - квантиль Стьюдента.

In [6]:
n = 400
p_hat = 170 / n
alpha = 0.91
p = (1 + alpha) / 2
t = st.t.ppf(p, df=n-1)

sd = p_hat - t * math.sqrt( p_hat * (1-p_hat) / n )
su = p_hat + t * math.sqrt( p_hat * (1-p_hat) / n )

print(f'{sd:.03f}, {su:.03f}')

0.383, 0.467


**Ответ:**

Лучший вариант охвата телезрителей - 46,7%

#### Задачи на построение доверительных интервалов

##### Задача №1

Глубина моря измеряется прибором, систематическая ошибка которого равна нулю, а случайные ошибки распределены нормально со средним отклонением 20м. Сколько надо сделать независимых измерений, чтобы определить глубину с ошибкой не более 15м при доверительной вероятности 0.9?

Постройте доверительный интервал для мат.ожидания ошибки измерения глубины и найдите требуемый объем выборки.
Формат ответа: через запятую с пробелом



*   найдите квантиль, участвующий в построении доверительного интервала и принимающий положительные значения
*   верхняя граница точности доверительного интервала
* требуемый объем выборки (наименьшее допустимое целое)




$ Решение: $

Систематическая ошибка равна нулю $ \Rightarrow \overline X = 0$ 

По условию известно, что $ x_i \sim N(a, \sigma^2) $. Усредним дисперсию

$ \overline X - a \sim N(0, \frac{\sigma^2}{n}) $

Вынесем дисперсию из под знака $N$

$ \sqrt{n} \frac{\overline X - a }{\sigma} \sim N(0,1) $

Для этого распределения мы знаем доверительный интервал.

$ \bigg ( \overline X \pm z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}}  \bigg ) $

В нашем случае, т.к. систематическая ошибка равна нулю

$ \bigg ( - z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}}, z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}}  \bigg ) $

Найдём квантиль.


In [ ]:
alpha = 0.9
p = (1 + alpha) / 2

z = st.norm.ppf(p)
z

1.6448536269514722

$ z_{\frac{1+\alpha}{2}} = 1.6448$

Верхняя граница точности указана в условии задачи и равна 15 м.

Найдём требуемый объем выборки, чтобы получить нужную точность. Точностью интервального оценивания называют половину довирительного интервала

$ \Delta = \frac{S^+ - S^-}{2}$

Отсюда выразим требуемое кол-во выборки

$ \Delta \geq \frac{S^+ - S^-}{2} = \frac{z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}} - (- z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}})}{2} = \frac{2 \cdot z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}}}{2} = z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}} $

Получили

$ \Delta \geq z_{\frac{1+\alpha}{2}} \frac{\sigma}{\sqrt{n}} $

Выразим $n$

$ n \geq \bigg ( \frac{z_{\frac{1+\alpha}{2}} \sigma}{\Delta}  \bigg )^2 $

Подставляем значения $ \Delta = 15$, $\sigma=20$, $ z_{\frac{1+\alpha}{2}} = 1.6448 $



In [ ]:
n = (z * 20 / 15)**2
n

4.809855029502956

Получили

$n \geq 4.8 \approx 5$ 

**Ответ:**

1. $ z_{\frac{1+\alpha}{2}} = 1.6448$
2. верхняя граница точности доверительного интервала = 15
3. $n = 5$

#### Таблица рекомендаций, какую статистику использовать

Таблица из [статьи](https://allfi.biz/CFA/Level-1/doveritelnyj-interval/).

| Распределение       | se ген. совокупности| Объем выборки  | Рекомендуемая методика |
| -----------------------:|-----------:| ----------:| --------------------:|
| нормальное              | известно   | маленький  | Z-оценка |
| нормальное              | известно   |  большой   | Z-оценка |
| нормальное              | неизвестно |  маленький | t-распределение |
| нормальное              | неизвестно |    большой | t-распределение или Z |
| отличное от нормального | известно   | маленький  | недоступно |
| отличное от нормального | известно   | большой    |  Z         |
| отличное от нормального | неизвестно | маленький  | недоступно |
| отличное от нормального | неизвестно | большой    |  t или Z   |
